### ISSUE TITLE
Failed build due to warnings in Murmur3 hash-fct

### ISSUE THREAD

**Issue by therealthingy** [hpcraink/fsprj2#30]
```bash
❯ rm -rf build && mkdir build &&  (cd build && cmake DCMAKE_BUILD_TYPE=Debug  -DENABLE_OUTPUT=LOGFILE -DWITH_MPI_IO=OFF -DENABLE_REMOTE_CONTROL=OFF -DFILENAME_RESOLUTION_ENABLED=ON -DHASH_FUNCTION=MD5HASH   ..  &&  cmake --build .)
-- The C compiler identification is GNU 11.3.0
-- The CXX compiler identification is GNU 11.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test HAVE_ATTRIBUTE_NONNULL
-- Performing Test HAVE_ATTRIBUTE_NONNULL - Success
-- Performing Test HAVE_ATTRIBUTE_ALIGNED
-- Performing Test HAVE_ATTRIBUTE_ALIGNED - Success
-- Performing Test HAVE_DECLSPEC_ALIGNED
-- Performing Test HAVE_DECLSPEC_ALIGNED - Failed
-- Performing Test HAVE_ATTRIBUTE_UNUSED
-- Performing Test HAVE_ATTRIBUTE_UNUSED - Success
-- Performing Test HAVE_ATTRIBUTE_CONSTRUCTOR
-- Performing Test HAVE_ATTRIBUTE_CONSTRUCTOR - Success
-- Performing Test HAVE_ATTRIBUTE_DESTRUCTOR
-- Performing Test HAVE_ATTRIBUTE_DESTRUCTOR - Success
-- Performing Test HAVE_ATTRIBUTE_THREAD
-- Performing Test HAVE_ATTRIBUTE_THREAD - Success
-- Performing Test HAVE_DECLSPEC_THREAD
-- Performing Test HAVE_DECLSPEC_THREAD - Failed
-- Performing Test HAVE_RECVMMSG_CONST_TIMESPEC
-- Performing Test HAVE_RECVMMSG_CONST_TIMESPEC - Failed
-- Performing Test HAVE_TIME_RDTSC
-- Performing Test HAVE_TIME_RDTSC - Success
-- Looking for include file stdbool.h
-- Looking for include file stdbool.h - found
-- Looking for include file stdlib.h
-- Looking for include file stdlib.h - found
-- Looking for include file stdint.h
-- Looking for include file stdint.h - found
-- Looking for include file sys/fcntl.h
-- Looking for include file sys/fcntl.h - found
-- Looking for include file sys/time.h
-- Looking for include file sys/time.h - found
-- Looking for include file sys/types.h
-- Looking for include file sys/types.h - found
-- Looking for include file sys/stat.h
-- Looking for include file sys/stat.h - found
-- Looking for include file sys/syscall.h
-- Looking for include file sys/syscall.h - found
-- Looking for include file unistd.h
-- Looking for include file unistd.h - found
-- Looking for include file pthread.h
-- Looking for include file pthread.h - found
-- Looking for include file aio.h
-- Looking for include file aio.h - found
-- Looking for include file getmnt.h
-- Looking for include file getmnt.h - not found
-- Looking for include file sched.h
-- Looking for include file sched.h - found
-- Looking for include file mntent.h
-- Looking for include file mntent.h - found
-- Looking for include file utmp.h
-- Looking for include file utmp.h - found
-- Looking for include file utmpx.h
-- Looking for include file utmpx.h - found
-- Looking for __open
-- Looking for __open - not found
-- Looking for __open64
-- Looking for __open64 - not found
-- Looking for __open_2
-- Looking for __open_2 - found
-- Looking for __open64_2
-- Looking for __open64_2 - found
-- Looking for open64
-- Looking for open64 - found
-- Looking for openat
-- Looking for openat - found
-- Looking for creat64
-- Looking for creat64 - found
-- Looking for pread
-- Looking for pread - found
-- Looking for pread64
-- Looking for pread64 - found
-- Looking for pwrite
-- Looking for pwrite - found
-- Looking for pwrite64
-- Looking for pwrite64 - found
-- Looking for lseek64
-- Looking for lseek64 - found
-- Looking for readv
-- Looking for readv - found
-- Looking for writev
-- Looking for writev - found
-- Looking for preadv
-- Looking for preadv - found
-- Looking for preadv64
-- Looking for preadv64 - found
-- Looking for pwritev
-- Looking for pwritev - found
-- Looking for pwritev64
-- Looking for pwritev64 - found
-- Looking for preadv2
-- Looking for preadv2 - found
-- Looking for preadv64v2
-- Looking for preadv64v2 - found
-- Looking for pwritev2
-- Looking for pwritev2 - found
-- Looking for pwritev64v2
-- Looking for pwritev64v2 - found
-- Looking for copy_file_range
-- Looking for copy_file_range - found
-- Looking for mmap
-- Looking for mmap - found
-- Looking for mmap64
-- Looking for mmap64 - found
-- Looking for munmap
-- Looking for munmap - found
-- Looking for msync
-- Looking for msync - found
-- Looking for mremap
-- Looking for mremap - found
-- Looking for fopen64
-- Looking for fopen64 - found
-- Looking for freopen64
-- Looking for freopen64 - found
-- Looking for fdopen
-- Looking for fdopen - found
-- Looking for fcloseall
-- Looking for fcloseall - found
-- Looking for flockfile
-- Looking for flockfile - found
-- Looking for ftrylockfile
-- Looking for ftrylockfile - found
-- Looking for funlockfile
-- Looking for funlockfile - found
-- Looking for fwide
-- Looking for fwide - found
-- Looking for fputc_unlocked
-- Looking for fputc_unlocked - found
-- Looking for fputwc_unlocked
-- Looking for fputwc_unlocked - found
-- Looking for putc_unlocked
-- Looking for putc_unlocked - found
-- Looking for putwc_unlocked
-- Looking for putwc_unlocked - found
-- Looking for fputs_unlocked
-- Looking for fputs_unlocked - found
-- Looking for fputws_unlocked
-- Looking for fputws_unlocked - found
-- Looking for putw
-- Looking for putw - found
-- Looking for fgetc_unlocked
-- Looking for fgetc_unlocked - found
-- Looking for fgetwc_unlocked
-- Looking for fgetwc_unlocked - found
-- Looking for getc_unlocked
-- Looking for getc_unlocked - found
-- Looking for getwc_unlocked
-- Looking for getwc_unlocked - found
-- Looking for getw
-- Looking for getw - found
-- Looking for getline
-- Looking for getline - found
-- Looking for getdelim
-- Looking for getdelim - found
-- Looking for fgets_unlocked
-- Looking for fgets_unlocked - found
-- Looking for fgetws_unlocked
-- Looking for fgetws_unlocked - found
-- Looking for fread_unlocked
-- Looking for fread_unlocked - found
-- Looking for fwrite_unlocked
-- Looking for fwrite_unlocked - found
-- Looking for fwprintf
-- Looking for fwprintf - found
-- Looking for vfwprintf
-- Looking for vfwprintf - found
-- Looking for fwscanf
-- Looking for fwscanf - found
-- Looking for vfscanf
-- Looking for vfscanf - found
-- Looking for vfwscanf
-- Looking for vfwscanf - found
-- Looking for feof_unlocked
-- Looking for feof_unlocked - found
-- Looking for ferror_unlocked
-- Looking for ferror_unlocked - found
-- Looking for clearerr_unlocked
-- Looking for clearerr_unlocked - found
-- Looking for ftello
-- Looking for ftello - found
-- Looking for ftello64
-- Looking for ftello64 - found
-- Looking for fseeko
-- Looking for fseeko - found
-- Looking for fseeko64
-- Looking for fseeko64 - found
-- Looking for fgetpos64
-- Looking for fgetpos64 - found
-- Looking for fsetpos64
-- Looking for fsetpos64 - found
-- Looking for fflush_unlocked
-- Looking for fflush_unlocked - found
-- Looking for setbuffer
-- Looking for setbuffer - found
-- Looking for setlinebuf
-- Looking for setlinebuf - found
-- Looking for madvise
-- Looking for madvise - found
-- Looking for posix_madvise
-- Looking for posix_madvise - found
-- Looking for sync
-- Looking for sync - found
-- Looking for syncfs
-- Looking for syncfs - found
-- Looking for fsync
-- Looking for fsync - found
-- Looking for fdatasync
-- Looking for fdatasync - found
-- Looking for fileno
-- Looking for fileno - found
-- Looking for dup3
-- Looking for dup3 - found
-- Looking for accept4
-- Looking for accept4 - found
-- Looking for pipe2
-- Looking for pipe2 - found
-- Looking for memfd_create
-- Looking for memfd_create - not found
-- Looking for tmpfile64
-- Looking for tmpfile64 - found
-- Looking for epoll_create
-- Looking for epoll_create - found
-- Looking for epoll_create1
-- Looking for epoll_create1 - found
-- Looking for mkstemp
-- Looking for mkstemp - found
-- Looking for mkostemp
-- Looking for mkostemp - found
-- Looking for mkstemps
-- Looking for mkstemps - found
-- Looking for mkostemps
-- Looking for mkostemps - found
-- Looking for eventfd
-- Looking for eventfd - found
-- Looking for inotify_init
-- Looking for inotify_init - found
-- Looking for inotify_init1
-- Looking for inotify_init1 - found
-- Looking for vfork
-- Looking for vfork - found
-- Looking for clone
-- Looking for clone - found
-- Looking for dirfd
-- Looking for dirfd - found
-- Looking for sendmmsg
-- Looking for sendmmsg - found
-- Looking for recvmmsg
-- Looking for recvmmsg - found
-- Looking for memrchr
-- Looking for memrchr - found
-- Looking for execvpe
-- Looking for execvpe - found
-- Looking for aio_read64
-- Looking for aio_read64 - found
-- Looking for aio_write64
-- Looking for aio_write64 - found
-- Looking for aio_init
-- Looking for aio_init - found
-- Looking for lio_listio64
-- Looking for lio_listio64 - found
-- Looking for aio_error64
-- Looking for aio_error64 - found
-- Looking for aio_return64
-- Looking for aio_return64 - found
-- Looking for aio_fsync64
-- Looking for aio_fsync64 - found
-- Looking for aio_suspend64
-- Looking for aio_suspend64 - found
-- Looking for aio_cancel64
-- Looking for aio_cancel64 - found
-- Looking for dlmopen
-- Looking for dlmopen - found
-- Looking for getmntent_r
-- Looking for getmntent_r - found
-- Looking for reallocarray
-- Looking for reallocarray - found
-- Looking for off64_t
-- Looking for off64_t - not found
-- Looking for O_CLOEXEC
-- Looking for O_CLOEXEC - found
-- Looking for O_DIRECTORY
-- Looking for O_DIRECTORY - found
-- Looking for O_NOFOLLOW
-- Looking for O_NOFOLLOW - found
-- Looking for O_TMPFILE
-- Looking for O_TMPFILE - found
-- Looking for O_DIRECT
-- Looking for O_DIRECT - found
-- Looking for O_NOATIME
-- Looking for O_NOATIME - found
-- Looking for O_PATH
-- Looking for O_PATH - found
-- Looking for O_LARGEFILE
-- Looking for O_LARGEFILE - found
-- Looking for RWF_APPEND
-- Looking for RWF_APPEND - found
-- Looking for MAP_32BIT
-- Looking for MAP_32BIT - found
-- Looking for MAP_ANONYMOUS
-- Looking for MAP_ANONYMOUS - found
-- Looking for MAP_DENYWRITE
-- Looking for MAP_DENYWRITE - found
-- Looking for MAP_EXECUTABLE
-- Looking for MAP_EXECUTABLE - found
-- Looking for MAP_FILE
-- Looking for MAP_FILE - found
-- Looking for MAP_GROWSDOWN
-- Looking for MAP_GROWSDOWN - found
-- Looking for MAP_HUGETLB
-- Looking for MAP_HUGETLB - found
-- Looking for MAP_HUGE_2MB
-- Looking for MAP_HUGE_2MB - not found
-- Looking for MAP_HUGE_1GB
-- Looking for MAP_HUGE_1GB - not found
-- Looking for MAP_LOCKED
-- Looking for MAP_LOCKED - found
-- Looking for MAP_NONBLOCK
-- Looking for MAP_NONBLOCK - found
-- Looking for MAP_NORESERVE
-- Looking for MAP_NORESERVE - found
-- Looking for MAP_POPULATE
-- Looking for MAP_POPULATE - found
-- Looking for MAP_STACK
-- Looking for MAP_STACK - found
-- Looking for MAP_UNINITIALIZED
-- Looking for MAP_UNINITIALIZED - not found
-- Looking for MADV_REMOVE
-- Looking for MADV_REMOVE - found
-- Looking for MADV_DONTFORK
-- Looking for MADV_DONTFORK - found
-- Looking for MADV_DOFORK
-- Looking for MADV_DOFORK - found
-- Looking for MADV_HWPOISON
-- Looking for MADV_HWPOISON - found
-- Looking for MADV_MERGEABLE
-- Looking for MADV_MERGEABLE - found
-- Looking for MADV_UNMERGEABLE
-- Looking for MADV_UNMERGEABLE - found
-- Looking for MADV_SOFT_OFFLINE
-- Looking for MADV_SOFT_OFFLINE - not found
-- Looking for MADV_HUGEPAGE
-- Looking for MADV_HUGEPAGE - found
-- Looking for MADV_NOHUGEPAGE
-- Looking for MADV_NOHUGEPAGE - found
-- Looking for MADV_DONTDUMP
-- Looking for MADV_DONTDUMP - found
-- Looking for MADV_DODUMP
-- Looking for MADV_DODUMP - found
-- Looking for MADV_FREE
-- Looking for MADV_FREE - found
-- Looking for MADV_WIPEONFORK
-- Looking for MADV_WIPEONFORK - found
-- Looking for MADV_KEEPONFORK
-- Looking for MADV_KEEPONFORK - found
-- Looking for SEEK_DATA
-- Looking for SEEK_DATA - found
-- Looking for SEEK_HOLE
-- Looking for SEEK_HOLE - found
-- Looking for HOST_NAME_MAX
-- Looking for HOST_NAME_MAX - found
-- Looking for _POSIX_HOST_NAME_MAX
-- Looking for _POSIX_HOST_NAME_MAX - found
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.2") 
-- Checking for module 'openssl'
--   Found openssl, version 3.0.2
-- Found Git: /usr/bin/git (found version "2.34.1") 
-- Looking for include file linux/kcmp.h
-- Looking for include file linux/kcmp.h - found
-- Found OpenMP_C: -fopenmp (found version "4.5") 
-- Found OpenMP_CXX: -fopenmp (found version "4.5") 
-- Found OpenMP: TRUE (found version "4.5")  
-- Found OpenMP_C: -fopenmp (found version "4.5") 
-- Found OpenMP_CXX: -fopenmp (found version "4.5") 
-- Configuring done
-- Generating done
-- Build files have been written to: /home/user/fsprj2/libiotrace/build
[  1%] Building C object src/libs/atomic_hash/CMakeFiles/atomic_hash_object.dir/atomic_hash.c.o
[  1%] Building C object src/libs/atomic_hash/CMakeFiles/atomic_hash_object.dir/hash_functions/hash_md5.c.o
/home/user/fsprj2/libiotrace/src/libs/atomic_hash/hash_functions/hash_md5.c: In function ‘md5hash’:
/home/user/fsprj2/libiotrace/src/libs/atomic_hash/hash_functions/hash_md5.c:8:3: error: ‘MD5’ is deprecated: Since OpenSSL 3.0 [-Werror=deprecated-declarations]
    8 |   MD5((unsigned char *)s, (size_t)len, (unsigned char *)r);
      |   ^~~
In file included from /home/user/fsprj2/libiotrace/src/libs/atomic_hash/hash_functions/hash_md5.c:2:
/usr/include/openssl/md5.h:52:38: note: declared here
   52 | OSSL_DEPRECATEDIN_3_0 unsigned char *MD5(const unsigned char *d, size_t n,
      |                                      ^~~
cc1: all warnings being treated as errors
gmake[2]: *** [src/libs/atomic_hash/CMakeFiles/atomic_hash_object.dir/build.make:90: src/libs/atomic_hash/CMakeFiles/atomic_hash_object.dir/hash_functions/hash_md5.c.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:465: src/libs/atomic_hash/CMakeFiles/atomic_hash_object.dir/all] Error 2
gmake: *** [Makefile:166: all] Error 2
```

**Comment by therealthingy** [comment]
UPDATE: MD5 has been replaced w/ Murmur3: https://github.com/hpcraink/fsprj2/commit/72ab8c8202b5ce0354b7213d8cb5310484c5a4ad

**Comment by therealthingy** [comment]
New error (since warnings are treated as errors): 

```
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c: In function ‘MurmurHash3_x64_128’:
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:26:5: error: switch missing default case [-Werror=switch-default]
   26 |     switch(len & 15) {
      |     ^~~~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:27:21: error: this statement may fall through [-Werror=implicit-fallthrough=]
   27 |         case 15: k2 ^= (uint64_t)(tail[14]) << 48;
      |                  ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:28:9: note: here
   28 |         case 14: k2 ^= (uint64_t)(tail[13]) << 40;
      |         ^~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:28:21: error: this statement may fall through [-Werror=implicit-fallthrough=]
   28 |         case 14: k2 ^= (uint64_t)(tail[13]) << 40;
      |                  ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:29:9: note: here
   29 |         case 13: k2 ^= (uint64_t)(tail[12]) << 32;
      |         ^~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:29:21: error: this statement may fall through [-Werror=implicit-fallthrough=]
   29 |         case 13: k2 ^= (uint64_t)(tail[12]) << 32;
      |                  ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:30:9: note: here
   30 |         case 12: k2 ^= (uint64_t)(tail[11]) << 24;
      |         ^~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:30:21: error: this statement may fall through [-Werror=implicit-fallthrough=]
   30 |         case 12: k2 ^= (uint64_t)(tail[11]) << 24;
      |                  ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:31:9: note: here
   31 |         case 11: k2 ^= (uint64_t)(tail[10]) << 16;
      |         ^~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:31:21: error: this statement may fall through [-Werror=implicit-fallthrough=]
   31 |         case 11: k2 ^= (uint64_t)(tail[10]) << 16;
      |                  ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:32:9: note: here
   32 |         case 10: k2 ^= (uint64_t)(tail[ 9]) << 8;
      |         ^~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:32:21: error: this statement may fall through [-Werror=implicit-fallthrough=]
   32 |         case 10: k2 ^= (uint64_t)(tail[ 9]) << 8;
      |                  ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:33:9: note: here
   33 |         case  9: k2 ^= (uint64_t)(tail[ 8]) << 0;
      |         ^~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:34:73: error: this statement may fall through [-Werror=implicit-fallthrough=]
   34 |                  k2 *= c2; k2 = ((k2 << 33) | (k2 >> 31)); k2 *= c1; h2 ^= k2;
      |                                                                      ~~~^~~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:35:9: note: here
   35 |         case  8: k1 ^= (uint64_t)(tail[ 7]) << 56;
      |         ^~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:35:21: error: this statement may fall through [-Werror=implicit-fallthrough=]
   35 |         case  8: k1 ^= (uint64_t)(tail[ 7]) << 56;
      |                  ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:36:9: note: here
   36 |         case  7: k1 ^= (uint64_t)(tail[ 6]) << 48;
      |         ^~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:36:21: error: this statement may fall through [-Werror=implicit-fallthrough=]
   36 |         case  7: k1 ^= (uint64_t)(tail[ 6]) << 48;
      |                  ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:37:9: note: here
   37 |         case  6: k1 ^= (uint64_t)(tail[ 5]) << 40;
      |         ^~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:37:21: error: this statement may fall through [-Werror=implicit-fallthrough=]
   37 |         case  6: k1 ^= (uint64_t)(tail[ 5]) << 40;
      |                  ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:38:9: note: here
   38 |         case  5: k1 ^= (uint64_t)(tail[ 4]) << 32;
      |         ^~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:38:21: error: this statement may fall through [-Werror=implicit-fallthrough=]
   38 |         case  5: k1 ^= (uint64_t)(tail[ 4]) << 32;
      |                  ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:39:9: note: here
   39 |         case  4: k1 ^= (uint64_t)(tail[ 3]) << 24;
      |         ^~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:39:21: error: this statement may fall through [-Werror=implicit-fallthrough=]
   39 |         case  4: k1 ^= (uint64_t)(tail[ 3]) << 24;
      |                  ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:40:9: note: here
   40 |         case  3: k1 ^= (uint64_t)(tail[ 2]) << 16;
      |         ^~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:40:21: error: this statement may fall through [-Werror=implicit-fallthrough=]
   40 |         case  3: k1 ^= (uint64_t)(tail[ 2]) << 16;
      |                  ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:41:9: note: here
   41 |         case  2: k1 ^= (uint64_t)(tail[ 1]) << 8;
      |         ^~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:41:21: error: this statement may fall through [-Werror=implicit-fallthrough=]
   41 |         case  2: k1 ^= (uint64_t)(tail[ 1]) << 8;
      |                  ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/user/repos/fsprj2/libiotrace/src/libs/atomic_hash/src/hash_functions/hash_murmur3.c:42:9: note: here
   42 |         case  1: k1 ^= (uint64_t)(tail[ 0]) << 0;
      |         ^~~~
cc1: all warnings being treated as errors
```

**Comment by therealthingy** [comment]
fixed w/ 8c0692ff0e8b72bbcc499405c02a17f0739ea481
